velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -28,51 +28,117 @@ namespace facebook::velox::exec::prefixsort {
 /// Provides encode/decode methods for PrefixSort.
 class PrefixSortEncoder {
  public:
-  /// 1. Only int64_t is supported now.
-  /// 2. Encoding is compatible with sorting ascending with no nulls.
+  /// Encode native primitive types(such as byte, int16, int32, int64, double,
+  /// float) to dest. There will be a separate method to handle string type as
+  /// follow-up.
+  /// 1. The first byte of the encode result is nullByte, which is determined by
+  ///    the nullsFirst :
+  ///    -If nullsFirst : invalid(means null) -> 0, valid(means not null) -> 1
+  ///    -If !nullsFirst : invalid(means null) -> 1, valid(means not null) -> 0
+  /// 2. The remaining bytes are the encoding result of value:
+  ///    -If isNull, we ignore the input value (this means the caller
+  ///     can pass any value, the reason for not using std::optional here is to
+  ///     avoid the overhead of box-unboxing) and set the remaining sizeof(T)
+  ///     bytes to '0', they do not affect the comparison results at all.
+  ///    -If !isNull, we set the remaining bytes as the result of
+  ///     encodeNoNulls, which will consider whether it is ascending or not. The
+  ///     result in descending order is equal to invert-bits(the result in
```

Comment:
I feel it is very important to make sure the encoding scheme is correct and that other readers can understand what it is and how it works. Would you create a Google doc to describe the encoding and explain how it works and why. It will be easier to iterate on the doc. Once the doc is looking good, we can transfer it into a documentation article in Velox or README.md.